Various improvements, feel free to cherry-pick

okurz opened the pull request:
* Add style checks with flake8,black,pylint
* Implement some flake style suggestions and fixes
* Consistently format python files with black
* README: Provide explanation and example for config values
* Add requirements file
* Add LICENSE
* README: Fix spelling
* README: Fix suggested use of python3 interpreter

sqozz commented on LICENSE:
could you comment on why MIT is the license you choose? Currently I lean more towards GPL and derivatives as they require proper documentation about changes (of e.g. code).

nicksinger commented on LICENSE:
could you comment on why MIT is the license you choose? Currently I lean more towards GPL and derivatives as they require proper documentation about changes (of e.g. code).

okurz replied:
> could you comment on why MIT is the license you choose?

Because it's less restrictive and for a (small) project like this one similar as to "scripts repos" that is my common choice to enfoster even easier collaboration and use.

> Currently I lean more towards GPL and derivatives as they require proper documentation about changes (of e.g. code).

What do you mean with "require proper documentation"? You mean that GPL requires downstream users to provide changes back in an open form? That's correct. So take GPL then :)

okurz replied:
Excluded all license changes for now

nicksinger commented on README.md:
```diff
@@ -34,7 +45,7 @@ the end of your password as it will break the login.
 # Execution
 
-Simply run `python downloader.py` and watch it fetch your documents.
+Simply run `python3 downloader.py` and watch it fetch your documents.
```
I understand that you might want to be explicit in the makefile but in documentation I don't see the need to mention an explicit version - especially since python2 is officially unsupported over 3 years now

okurz replied:
Yes, correct. But on multiple Linux distributions that means that "python" will either not exist or still be the outdated not anymore upstream supported python 2.

okurz replied:
excluded that change from this PR for now

nicksinger commented on Makefile:
```diff
@@ -0,0 +1,20 @@
+
+.PHONY: test-online
+test-online:
+	python3 ./downloader.py
```
why do you consider this a "test"? Most likely it would just fail because we cannot define sane defaults for a proper functional test. I think this is misleading and should not be included.

okurz replied:
Sure. As A stated in the subject, feel free to cherry pick. But I can also create pull requests one by one for the parts that you do like

okurz replied:
excluded this for now from this PR

nicksinger commented on Makefile:
```diff
@@ -0,0 +1,20 @@
+
+.PHONY: flake8
+flake8:
+	flake8 $$(git ls-files "*.py")
```
Could the git call be replaced with `find . -maxdepth 1 -name "*.py"`? Otherwise the style-check only works if the repository was cloned via git

okurz replied:
Yeah, we have similar approaches all over the places like https://github.com/os-autoinst/openQA/blob/master/Makefile#L318 so how about

```suggestion
	flake8 $$(git ls-files "*.py" 2>/dev/null ||find . -maxdepth 1 -name "*.py")
```

nicksinger commented on adp.py:
```diff
@@ -66,10 +68,9 @@ def __init__(self):
             (self.api_endpoint.scheme, self.api_endpoint.netloc, "csrf", "", "", "")
         )
-        csrf_req = self.websession.get(csrf_url)
+        _ = self.websession.get(csrf_url)
```
```python
self.websession.get(csrf_url) # the webui does the same so it might be needed to satisfy the api endpoints
```

nicksinger replied:
I wanted to use the fancy "code suggestion" but have not enough github experience :) The assignment here is useless (and leftover from testing) and this is why I suggested only a comment explaining what happens instead because we don't actually need the return-value. The `_`-syntax I usually know from places where an assignment is necessary, e.g. in for-loops

okurz replied:
Using your suggested version now

nicksinger reviewed (changes requested):
Thank you very much for this great housekeeping and stuff I always wanted to do but was too lazy to care :) I added a few changes I'd like to see but if they are implemented I could merge this completely. If you exclude the LICENSE I can think a little more about it and add it later but I can understand if you want your changes properly covered under a FOSS license.

okurz commented:
> Thank you very much for this great housekeeping and stuff I always wanted to do but was too lazy to care :) I added a few changes I'd like to see but if they are implemented I could merge this completely. If you exclude the LICENSE I can think a little more about it and add it later but I can understand if you want your changes properly covered under a FOSS license.

I excluded the debated changes and implemented changes from this PR. Will create another PR on top for the delta for further discussion

nicksinger commented:
awesome, thank you very much!
